zowex/src/comm.rs

```diff
@@ -261,23 +284,21 @@ pub fn comm_talk(message: &[u8], stream: &mut DaemonClient) -> io::Result<i32> {
                     break;
                 }
             }
+            Err(e) if e.kind() == io::ErrorKind::WouldBlock => {
+                // The daemon should try to read again, as we cannot block here
+                continue;
+            }
             Err(err_val) => return Err(err_val),
         } // end match on read
     } // end loop
 
     // Terminate connection. Ignore NotConnected errors returned on macOS.
     // https://doc.rust-lang.org/std/net/struct.TcpStream.html#method.shutdown
     #[cfg(target_family = "unix")]
-    match stream.shutdown(Shutdown::Read) {
+    match reader.shutdown().await {
         Err(ref e) if e.kind() == io::ErrorKind::NotConnected => (),
         result => result?,
     }
-    #[cfg(target_family = "unix")]
```

Comment:
As I recall, this step was added to avoid some weird error about connection terminated on some platform. If we have confirmed that the extra shutdown is no longer needed anywhere, then all is good.

Reply:
I updated the import to use `tokio::io::BufReader` for the read stage, so I can still access the functions from the stream object. I was going to keep this call, but since `stream` was moved when initializing a `BufReader`, I no longer could use it. `reader.shutdown` will end up calling `stream.shutdown`, so that functionality still remains 🙂